From db72f7a6ec35cad8430b06c929919a8e49324e43 Mon Sep 17 00:00:00 2001
From: Markus Niebel <Markus.Niebel@ew.tq-group.com>
Date: Thu, 24 Oct 2024 16:55:34 +0200
Subject: [PATCH] Add generic wrapper for mkimage_imx8

Add a wrapper that tees the stderr output to a log file, allowing for
more robust parsing for CSF / SRK / HAB block offsets.

Signed-off-by: Markus Niebel <Markus.Niebel@ew.tq-group.com>
---
 Makefile        | 14 +++++++-------
 iMX8DXL/soc.mak |  2 +-
 iMX8QX/soc.mak  |  2 +-
 iMX8ULP/soc.mak |  2 +-
 iMX91/soc.mak   |  2 +-
 iMX93/soc.mak   |  2 +-
 iMX95/soc.mak   |  2 +-
 mkimage_imx8    |  7 +++++++
 8 files changed, 20 insertions(+), 13 deletions(-)
 create mode 100755 mkimage_imx8

diff --git a/Makefile b/Makefile
index e4cf1c3..329c0d6 100644
--- a/Makefile
+++ b/Makefile
@@ -1,5 +1,5 @@
 
-MKIMG = $(PWD)/mkimage_imx8
+MKIMG_REAL = $(PWD)/mkimage_imx8.real
 CC = gcc
 CFLAGS ?= -g -O2 -Wall -std=c99 -static
 INCLUDE += $(CURR_DIR)/src
@@ -30,10 +30,10 @@ else
 endif
 
 #print out usage as the default target
-all: $(MKIMG) help
+all: $(MKIMG_REAL) help
 
 clean:
-	@rm -f $(MKIMG)
+	@rm -f $(MKIMG_REAL)
 	@rm -f src/build_info.h
 ifdef O
 	@$(MAKE) --no-print-directory -C $(O) -f ../$(SOC_DIR)/soc.mak clean
@@ -47,11 +47,11 @@ endif
 	@$(MAKE) --no-print-directory -C iMX91 -f soc.mak  clean
 	@$(MAKE) --no-print-directory -C iMX95 -f soc.mak  clean
 
-$(MKIMG): src/build_info.h $(SRCS)
-	@echo "Compiling mkimage_imx8"
-	$(CC) $(CFLAGS) $(SRCS) -o $(MKIMG) -I src
+$(MKIMG_REAL): src/build_info.h $(SRCS)
+	@echo "Compiling $(MKIMG_REAL)"
+	$(CC) $(CFLAGS) $(SRCS) -o $@ -I src
 
-bin: $(MKIMG)
+bin: $(MKIMG_REAL)
 
 src/build_info.h:
 	@echo -n '#define MKIMAGE_COMMIT 0x' > src/build_info.h
diff --git a/iMX8DXL/soc.mak b/iMX8DXL/soc.mak
index f0ab34b..231bd09 100755
--- a/iMX8DXL/soc.mak
+++ b/iMX8DXL/soc.mak
@@ -62,7 +62,7 @@ Image1: Image
 
 .PHONY: clean nightly
 clean:
-	@rm -f $(MKIMG) u-boot-atf-container.img Image0 Image1 u-boot-hash.bin u-boot-atf.bin head.hash u-boot-atf-container.img flash.bin
+	@rm -f $(MKIMG).real u-boot-atf-container.img Image0 Image1 u-boot-hash.bin u-boot-atf.bin head.hash u-boot-atf-container.img flash.bin
 	@rm -rf extracted_imgs
 	@echo "imx8dxl clean done"
 
diff --git a/iMX8QX/soc.mak b/iMX8QX/soc.mak
index f2f7849..8f19a57 100755
--- a/iMX8QX/soc.mak
+++ b/iMX8QX/soc.mak
@@ -63,7 +63,7 @@ Image1: Image
 
 .PHONY: clean nightly
 clean:
-	@rm -f $(MKIMG) $(DCD_CFG) .imx8qx_dcd.cfg.cfgtmp.d u-boot-atf-container.img Image0 Image1
+	@rm -f $(MKIMG).real $(DCD_CFG) .imx8qx_dcd.cfg.cfgtmp.d u-boot-atf-container.img Image0 Image1
 	@rm -rf extracted_imgs
 	@echo "imx8qx clean done"
 
diff --git a/iMX8ULP/soc.mak b/iMX8ULP/soc.mak
index ef74ecd..d4bee4d 100644
--- a/iMX8ULP/soc.mak
+++ b/iMX8ULP/soc.mak
@@ -58,7 +58,7 @@ u-boot-atf-container.img: bl31.bin u-boot-hash.bin
 
 .PHONY: clean nightly
 clean:
-	@rm -f $(MKIMG) u-boot-atf-container.img
+	@rm -f $(MKIMG).real u-boot-atf-container.img
 	@rm -rf extracted_imgs
 	@echo "imx8ulp clean done"
 
diff --git a/iMX91/soc.mak b/iMX91/soc.mak
index b87a5be..25fd7d2 100644
--- a/iMX91/soc.mak
+++ b/iMX91/soc.mak
@@ -160,7 +160,7 @@ flash_fw.bin: FORCE
 
 .PHONY: clean nightly
 clean:
-	@rm -f $(MKIMG) u-boot-atf-container.img u-boot-spl-ddr.bin u-boot-spl-ddr-qb.bin u-boot-hash.bin
+	@rm -f $(MKIMG).real u-boot-atf-container.img u-boot-spl-ddr.bin u-boot-spl-ddr-qb.bin u-boot-hash.bin
 	@rm -rf extracted_imgs
 	@echo "imx91 clean done"
 
diff --git a/iMX93/soc.mak b/iMX93/soc.mak
index 9421904..d0679ff 100644
--- a/iMX93/soc.mak
+++ b/iMX93/soc.mak
@@ -161,7 +161,7 @@ flash_fw.bin: FORCE
 
 .PHONY: clean nightly
 clean:
-	@rm -f $(MKIMG) u-boot-atf-container.img u-boot-spl-ddr.bin u-boot-spl-ddr-qb.bin u-boot-hash.bin
+	@rm -f $(MKIMG).real u-boot-atf-container.img u-boot-spl-ddr.bin u-boot-spl-ddr-qb.bin u-boot-hash.bin
 	@rm -rf extracted_imgs
 	@echo "imx93 clean done"
 
diff --git a/iMX95/soc.mak b/iMX95/soc.mak
index 4cce38e..9940e66 100644
--- a/iMX95/soc.mak
+++ b/iMX95/soc.mak
@@ -265,7 +265,7 @@ flash_fw.bin: FORCE
 
 .PHONY: clean nightly
 clean:
-	@rm -f $(MKIMG) u-boot-atf-container.img u-boot-spl-ddr-v2.bin m33-oei-ddrfw.bin a55-oei-ddrfw.bin u-boot-hash.bin flash.bin head.hash boot-spl-container.img
+	@rm -f $(MKIMG).real u-boot-atf-container.img u-boot-spl-ddr-v2.bin m33-oei-ddrfw.bin a55-oei-ddrfw.bin u-boot-hash.bin flash.bin head.hash boot-spl-container.img
 	@rm -rf extracted_imgs
 	@echo "imx95 clean done"
 
diff --git a/mkimage_imx8 b/mkimage_imx8
new file mode 100755
index 0000000..834aba9
--- /dev/null
+++ b/mkimage_imx8
@@ -0,0 +1,7 @@
+#!/bin/sh
+
+OUTLOG=mkimage.log
+SCRIPTPATH=$(dirname "$(readlink -f "${0}")")
+
+
+exec "${SCRIPTPATH}/mkimage_imx8.real" "$@" 2>&1 | tee "${OUTLOG}" >&2
